main.go

```diff
@@ -37,7 +35,20 @@ const (
 // Record represents a single export row in the export file
 type Record map[string]interface{}
 
-type ExportProcessor func(warehouse.Warehouse, []string, *client.Client, []client.ExportMeta) (int, error)
+type HauserService struct {
+	config       *config.Config
+	fsClient     client.DataExportClient
+	warehouse    warehouse.Warehouse
+	tableColumns []string
+}
+
+func NewHauser(config *config.Config, fsClient client.DataExportClient, warehouse warehouse.Warehouse) *HauserService {
```

Comment:
While we are refactoring here, should we move `HauserService` out of `main.go` into its own file or package?

Reply:
I can do that. Do you think we should move it to an `internal` package? Or just something like "core"?

Reply:
`internal/service.go` seems reasonable to me!